### PR TITLE
Print hint message before invoking installation scripts and fix a bunch of shell check errors

### PR DESCRIPTION
**Pull request by m1093782566**
Per @sdake 

> a command line option could be passed to install_prereqs.sh that disables the "are you sure you want to continue?" feature of each script and default to "ask for permission".

In tests/e2e/local scripts, we usually install/update packages with the option `--quiet`, probably we need to print hint message before invoking installation scripts.

@sdake you want this?

### CODE REVIEW

File: tests/e2e/local/minikube/install_prereqs.sh [hunk 1]
```diff
@@ -1,12 +1,27 @@
 #!/bin/bash
 
 case "${OSTYPE}" in
-  darwin*) sh install_prereqs_macos.sh;;
```
Comment (hklai) [inline tests/e2e/local/minikube/install_prereqs.sh]: Can we refactor these lines and reuse them as a function?

Comment (m1093782566) [reply tests/e2e/local/minikube/install_prereqs.sh]: Sure. I just pushed an update, PTAL. Thanks!

**Comment by hklai** [comment]
@m1093782566 could you rebase and make sure shellcheck is not broken again by this? Thanks!

**Comment by m1093782566** [comment]
Sure, @hklai 

Rebased to see test result...

**Comment by m1093782566** [comment]
Also /cc @sebastienvas

Per @hklai, I just pushed an update to fix the shell check errors in `tests/upgrage/test_upgrade_from.sh`

File: tests/upgrade/test_upgrade_from.sh [hunk 2]
```diff
@@ -208,26 +208,26 @@ sleep 100
...
     echo "No Code 200 found in internal traffic log"
 elif [[ ${pod_log_str} != *"(100.0"* ]]; then
-    printf "\n\nErrors found in internal traffic log:\n\n"
+    printf "\\nErrors found in internal traffic log:\\n"
     cat ${POD_FORTIO_LOG}
     exit 1
 fi
 
-popd
+popd || exit
```
Comment (sebastienvas) [inline tests/upgrade/test_upgrade_from.sh]: why ?

File: tests/upgrade/test_upgrade_from.sh [hunk 3]
```diff
@@ -183,9 +183,9 @@ kubectl create namespace istio-system
 kubectl create namespace ${TEST_NAMESPACE}
 kubectl label namespace ${TEST_NAMESPACE} istio-injection=enabled
 
-pushd ${ISTIO_ROOT}
+pushd "${ISTIO_ROOT}" || exit
```
Comment (sebastienvas) [inline tests/upgrade/test_upgrade_from.sh]: why ? should you do exit 1 ?

File: tests/upgrade/test_upgrade_from.sh [hunk 4]
```diff
@@ -35,74 +35,74 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
...
     writeMsg "Sending external traffic"
-    fortio load -c 32 -t 300s -qps 10 -H "Host:echosrv.test.svc.cluster.local" http://${1}/echo?size=200 &> ${LOCAL_FORTIO_LOG}
+    fortio load -c 32 -t 300s -qps 10 -H "Host:echosrv.test.svc.cluster.local" http://"${1}"/echo?size=200 &> ${LOCAL_FORTIO_LOG}
 }
 
 restartDataPlane() {
     # Apply label within deployment spec.
     # This is a hack to force a rolling restart without making any material changes to spec.
-    cur_time=$(date +%s)
+    # cur_time=$(date +%s)
```
Comment (sebastienvas) [inline tests/upgrade/test_upgrade_from.sh]: if it is not used, then remove it.

File: tests/upgrade/test_upgrade_from.sh [hunk 5]
```diff
@@ -35,74 +35,74 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
...
-   installIstioSystemAtVersionHelmTemplate ${1} ${2}
+   gitChangeToTag "${2}"
+   installIstioSystemAtVersionHelmTemplate "${1}" "${2}"
    gitRestorePrevBranch
 }
 
 installTest() {
     writeMsg "Installing test deployments"
-   kubectl apply -n ${TEST_NAMESPACE} -f ${ISTIO_ROOT}/tests/upgrade/templates/gateway.yaml
+   kubectl apply -n ${TEST_NAMESPACE} -f "${ISTIO_ROOT}"/tests/upgrade/templates/gateway.yaml
```
Comment (sebastienvas) [inline tests/upgrade/test_upgrade_from.sh]: prefer "${ISTIO_ROOT}/tests/upgrade/templates/gateway.yaml"

File: tests/upgrade/test_upgrade_from.sh [hunk 6]
```diff
@@ -35,74 +35,74 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
...
 installIstioSystemAtVersionHelmTemplate() {
-   helm template ${ISTIO_ROOT}/install/kubernetes/helm/istio \
+   helm template "${ISTIO_ROOT}"/install/kubernetes/helm/istio \
     --name istio --namespace istio-system \
     --set gateways.istio-ingressgateway.replicaCount=4 \
     --set gateways.istio-ingressgateway.autoscaleMin=4 \
-    --set global.hub=${1} \
-    --set global.tag=${2} > ${ISTIO_ROOT}/istio.yaml
+    --set global.hub="${1}" \
+    --set global.tag="${2}" > "${ISTIO_ROOT}"/istio.yaml
```
Comment (sebastienvas) [inline tests/upgrade/test_upgrade_from.sh]: "${ISTIO_ROOT}/istio.yaml"

**Pull request merged**